sechub-server/src/test/java/com/daimler/sechub/domain/schedule/ScheduleJobMarkerServiceTest.java

```diff
@@ -13,62 +13,63 @@
 
-	private ScheduleJobMarkerService serviceToTest;
+    private ScheduleJobMarkerService serviceToTest;
 
-	@Before
-	public void before() throws Exception {
-		serviceToTest = new ScheduleJobMarkerService();
+    @Before
+    public void before() throws Exception {
+        serviceToTest = new ScheduleJobMarkerService();
+        factory = mock(SchedulerStrategyFactory.class);
+        strategy = mock(FirstComeFirstServeSchedulerStrategy.class);
+        jobRepository = mock(SecHubJobRepository.class);
+        
+        uuid = UUID.randomUUID();
 
-		uuid = UUID.randomUUID();
+        serviceToTest.jobRepository = jobRepository;
+        serviceToTest.schedulerStrategyFactory = factory;
+        strategy.jobRepository = jobRepository;
 
-		jobRepository = mock(SecHubJobRepository.class);
+        secHubJob = mock(ScheduleSecHubJob.class);
 
-		serviceToTest.jobRepository = jobRepository;
-		
-		secHubJob = mock(ScheduleSecHubJob.class);
-		
-		when(secHubJob.getUUID()).thenReturn(uuid);
-	}
+        when(factory.build()).thenReturn(strategy);
+        when(strategy.nextJobId()).thenReturn(uuid);
+        when(jobRepository.getJob(any())).thenReturn(Optional.of(secHubJob));
+    }
 
-	@Test
-	public void markNextJobExecutedByThisPOD__calls_jobrepository_findNextJobToExecute() throws Exception {
-		/* prepare */
-		ScheduleSecHubJob job = mock(ScheduleSecHubJob.class);
-		when(job.getUUID()).thenReturn(UUID.randomUUID());
-		when(jobRepository.findNextJobToExecute()).thenReturn(Optional.of(job));
+    @Test
+    public void markNextJobExecutedByThisPOD__calls_jobrepository_getjob_executed() throws Exception {
+        /* execute */
+        serviceToTest.markNextJobToExecuteByThisInstance();
 
-		/* execute */
-		serviceToTest.markNextJobExecutedByThisPOD();
+        /* test */
+        verify(jobRepository).getJob(uuid);
+    }
 
-		/* test */
-		verify(jobRepository).findNextJobToExecute();
-	}
-	
-	@Test
-	public void markNextJobExecutedByThisPOD__updates_execution_state_to_started() throws Exception {
-		/* prepare */
-		ScheduleSecHubJob job = mock(ScheduleSecHubJob.class);
-		when(job.getUUID()).thenReturn(UUID.randomUUID());
-		when(jobRepository.findNextJobToExecute()).thenReturn(Optional.of(job));
-		when(jobRepository.save(job)).thenReturn(job);
+    @Test
+    public void markNextJobExecutedByThisPOD__updates_execution_state_to_started() throws Exception {
+        /* prepare */
+        when(jobRepository.save(secHubJob)).thenReturn(secHubJob);
 
-		/* execute */
-		ScheduleSecHubJob result = serviceToTest.markNextJobExecutedByThisPOD();
+        /* execute */
+        ScheduleSecHubJob result = serviceToTest.markNextJobToExecuteByThisInstance();
 
-		/* test */
-		verify(job).setStarted(any());
-		verify(job).setExecutionState(eq(ExecutionState.STARTED));
-		
-		assertEquals(job,result);
-	}
+        /* test */
+        verify(result).setStarted(any());
+        verify(result).setExecutionState(eq(ExecutionState.STARTED));
```

Comment:
The implementation before did check if the persisted job was changed.
Here the result is now checked for the change.
Because having the same mock object the test is green.

But maybe it's more clear when we would verify on sechubJob instance? But it's your choice, maybe it's better readable/ clear for you.

Reply:
done